From 86f351f5ec08badbcdf417eb582dec0579dd07b8 Mon Sep 17 00:00:00 2001
From: Yunzhi Li <lyz@rock-chips.com>
Date: Thu, 15 Oct 2015 11:51:04 +0800
Subject: [PATCH] usb: rk3368: set disconnect threshold to 625 mV

The disconnect threshold of 28nm usb phy from innosilicon is
imprecise, if the value set to 575 mV will trigger unnecessary
disconnect interrupt.

Change-Id: I63985fb9257a1142bdad476583c0c141de70618d
Signed-off-by: Yunzhi Li <lyz@rock-chips.com>
---
 drivers/usb/dwc_otg_310/usbdev_rk3368.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/usb/dwc_otg_310/usbdev_rk3368.c b/drivers/usb/dwc_otg_310/usbdev_rk3368.c
index dd5771d228e0..7af3c5488349 100644
--- a/drivers/usb/dwc_otg_310/usbdev_rk3368.c
+++ b/drivers/usb/dwc_otg_310/usbdev_rk3368.c
@@ -23,8 +23,8 @@ static void usb20otg_hw_init(void)
 	/* Turn off differential receiver in suspend mode */
 	uoc_write(UOC_HIWORD_UPDATE(0, 1, 2), 0x798);
 
-	/* Set disconnect detection trigger point to 600mv */
-	uoc_write(UOC_HIWORD_UPDATE(0, 0xf, 11), 0x79c);
+	/* Set disconnect detection trigger point to 625mv */
+	uoc_write(UOC_HIWORD_UPDATE(0x9, 0xf, 11), 0x79c);
 
 	/* other haredware init,include:
 	 * DRV_VBUS GPIO init */
@@ -220,8 +220,8 @@ static void usb20ehci_hw_init(void)
 {
 	/* Turn off differential receiver in suspend mode */
 	uoc_write(UOC_HIWORD_UPDATE(0, 1, 2), 0x7b8);
-	/* Set disconnect detection trigger point to 600mv */
-	uoc_write(UOC_HIWORD_UPDATE(1, 0xf, 11), 0x7bc);
+	/* Set disconnect detection trigger point to 625mv */
+	uoc_write(UOC_HIWORD_UPDATE(0x9, 0xf, 11), 0x7bc);
 
 	/* other haredware init,include:
 	 * DRV_VBUS GPIO init */
-- 
2.35.3

